src/pid/PIDControllerRunner.java

```diff
@@ -0,0 +1,31 @@
+package pid;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.IMotorControllerEnhanced;
+
+public class PIDControllerRunner implements PIDRunner {
+  private IMotorControllerEnhanced motorControllerEnhancedPIDRunner;
+  private MotorControllerEnhancedPIDController motorControllerEnhancedPIDController;
+
+  public PIDControllerRunner(IMotorControllerEnhanced MotorControllerEnhancedPIDRunner,
+                             MotorControllerEnhancedPIDController motorControllerEnhancedPIDController) {
+    this.motorControllerEnhancedPIDRunner = MotorControllerEnhancedPIDRunner;
+    this.motorControllerEnhancedPIDController = motorControllerEnhancedPIDController;
+  }
+
+  public void startPIDLoop() {
+    motorControllerEnhancedPIDController.resetAccumulator();
+    motorControllerEnhancedPIDRunner.set(motorControllerEnhancedPIDRunner.getControlMode(),
+        motorControllerEnhancedPIDController.calculate());
+  }
+
+  public void terminatePIDLoop(int remainOrStop) {
```

Comment:
Can't see why you've changed it from stop to terminate (besides sounding cooler 🆒 ).  Start/stop are opposites and it helps understanding that those functions are opposites too.

Reply:
[https://www.google.com/search?q=sad+face&rlz=1C1EKKP_enIL818IL818&oq=sad+face&aqs=chrome..69i57j0l7.2123j0j7&sourceid=chrome&ie=UTF-8](url)

Reply:
page no found, but I assume from url it's a 😢